File: python/program/Garage.py
class Garage:
    def __init__(self, nama, luas):
        self._nama = nama
        self._luas = luas
        self._vehicles = []

    def get_nama(self):
        return self._nama

    def get_luas(self):
        return self._luas

    def set_nama(self, nama):
        self._nama = nama

    def set_luas(self, luas):
        self._luas = luas

    def parkir_kendaraan(self, vehicle):
        self._vehicles.append(vehicle)
        print(f"Berhasil masukin mobil ke garasi {self._nama}.")

    def keluarkan_kendaraan(self, vehicle):
        if vehicle in self._vehicles:
            self._vehicles.remove(vehicle)
            print(f"Berhasil keluarin mobil ke garasi {self._nama}.")
            return vehicle
        else:
            print("Kendaraan tidak ditemukan.")

File: cpp/program/Garage.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include "Vehicle.h"

using namespace std;

class Garage {
private:
    string nama;
    int luas;
    vector<Vehicle*> vehicles;

public:
    Garage(string nama, int luas) : nama(nama), luas(luas) {}

    string getNama() const {
        return nama;
    }

    int getLuas() const {
        return luas;
    }

    vector<Vehicle*> getVehicles() const {
        return vehicles;
    }

    void setNama(const string& newNama) {
        nama = newNama;
    }

    void setLuas(int newLuas) {
        luas = newLuas;
    }

    void parkirKendaraan(Vehicle* vehicle) {
        vehicles.push_back(vehicle);
        cout << "Berhasil masukin mobil ke garasi " << nama << "." << endl;
    }

    bool keluarkanKendaraan(Vehicle* vehicle) {
        for (auto it = vehicles.begin(); it != vehicles.end(); ++it) {
            if (*it == vehicle) {
                vehicles.erase(it);
                cout << "Berhasil keluarin mobil dari garasi " << nama << "." << endl;
                return true;
            }
        }
        cout << "Kendaraan tidak ditemukan." << endl;
        return false;
    }
};
